lang/cs/Org.Apache.REEF.Network/NetworkService/WritableNsMessage.cs

```diff
@@ -101,12 +101,13 @@ public void Read(IDataReader reader)
             SourceId = _factory.Create(reader.ReadString());
             DestId = _factory.Create(reader.ReadString());
             int messageCount = reader.ReadInt32();
+            string dataType = reader.ReadString();
 
             Data = new List<T>();
 
             for (int index = 0; index < messageCount; index++)
             {
-                var dataPoint = (T)_injection.ForkInjector().GetInstance(typeof(T));
+                var dataPoint = (T)_injection.ForkInjector().GetInstance(Type.GetType(dataType));
```

Comment:
Is the original typeof<T> doesn't work? What is the type that is allowed? Any type? What is the type string passed in?